quic/s2n-quic-core/src/recovery/bbr/full_pipe.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    counter::{Counter, Saturating},
+    number::Fraction,
+    recovery::{bandwidth, bandwidth::Bandwidth, bbr},
+};
+
+/// Estimator for determining if BBR has fully utilized its available bandwidth ("filled the pipe")
+#[derive(Debug, Default, Clone)]
+pub(crate) struct Estimator {
+    //= https://tools.ietf.org/id/draft-cardwell-iccrg-bbr-congestion-control-02#2.13
+    //# A boolean that records whether BBR estimates that it has ever
+    //# fully utilized its available bandwidth ("filled the pipe").
+    filled_pipe: bool,
+    //= https://tools.ietf.org/id/draft-cardwell-iccrg-bbr-congestion-control-02#2.13
+    //# A recent baseline BBR.max_bw to estimate if BBR has "filled the pipe" in Startup.
+    full_bw: Bandwidth,
+    //= https://tools.ietf.org/id/draft-cardwell-iccrg-bbr-congestion-control-02#2.13
+    //# The number of non-app-limited round trips without large increases in BBR.full_bw.
+    full_bw_count: Counter<u8, Saturating>,
+    /// The number of discontiguous bursts of lost packets in the last round
+    loss_bursts: Counter<u8, Saturating>,
+    /// True if BBR was in fast recovery in the last round
+    in_recovery_last_round: bool,
+}
+
+impl Estimator {
+    /// Returns true if BBR estimates that is has ever fully utilized its available bandwidth
+    #[allow(dead_code)] // TODO: Remove when used
+    #[inline]
+    pub(crate) fn filled_pipe(&self) -> bool {
```

Comment:
We could probably get away with `pub` rather than `pub(crate)` since the struct itself is `pub(crate)`